### ISSUE TITLE
How to yarn/npm link pixi.js into a webpack 

### ISSUE THREAD

**Issue by georeith** [pixijs/pixi.js#6530]
This is not a bug request as such but is for anyone else who ends up searching issues for this.

When you `yarn link` `pixi.js` into a webpack built application you will likely come across the error:

> Cannot read property 'start' of undefined

This is because of the way `node_modules` are handled in the monorepo. Where each package contains a `node_modules` folder which symlinks to the other packages in the monorepo it requires. Because each package ends up with its own symlink webpack loads each import for each module again.

The simplest solution I found to this is adding the following `resolve.alias` config to `webpack`:

``` js
{ '@pixi': 'pixi.js/node_modules/@pixi' }
```

**Note**: This could be resolved by using yarn workspaces to manage the pixi.js monorepo which hoist monorepo imports to top level `node_modules` to prevent this.

**Comment by ivanpopelyshev** [comment]
Yeah, not possible to just link `pixi.js` to the project like I did with v4. That's a problem.

**Comment by bigtimebuddy** [comment]
@ SukantPal didn’t you recently stub you toe on this too? Did you find a workaround?

**Comment by SukantPal** [comment]
@bigtimebuddy I didn't see this problem per say. My local template's Webpack config is:

```js
const HtmlWebPackPlugin = require('html-webpack-plugin');
const CopyWebpackPlugin = require('copy-webpack-plugin');
const path = require('path');

module.exports = {
    mode: 'development',
    entry: './src/index.js',
    output: {
        path: path.resolve(__dirname, 'dist'),
        filename: 'app.bundle.js',
    },
    module: {
        rules: [
            {
                test: /\.(js)$/,
                exclude: /node_modules/,
                use: {
                    loader: 'babel-loader',
                },
            },
            {
                test: /\.html$/,
                use: [
                    {
                        loader: 'html-loader',
                    },
                ],
            },
            {
                test: /\.jpeg$/,
                use: {
                    loader: 'file-loader',
                },
            },
        ],
    },
    plugins: [
        new HtmlWebPackPlugin({
            template: './src/index.html',
            filename: './index.html',
        }),
        new CopyWebpackPlugin([{
            from: './src/assets', to: './assets',
        }]),
    ],
    watchOptions: {
        aggregateTimeout: 300,
        ignored: ['/node_modules/(?!pixi.js)/'],
    },
};
```

I don't see anything special here.

**Comment by SukantPal** [comment]
@ivanpopelyshev  @bigtimebuddy I remember now, I couldn't link pixi.js using `npm link`. So I import as:

```js
const PIXI = require('../../pixi.js/bundles/pixi.js/lib/pixi');
import * as PIXI from '../../pixi.js/bundles/pixi.js/lib/pixi.es';
```

**Comment by georeith** [comment]
> I don't see anything special here.

This may be due to my [`resolve.symlinks`](https://webpack.js.org/configuration/resolve/#resolvesymlinks) settings then . I have it set to `false`.

**Comment by SukantPal** [comment]
I created pixi-fiddle to make it easier for new developers to work on PixiJS. You can clone this repo (https://github.com/SukantPal/pixi-fiddle).

* Create a `fiddle.config.json` file and add a `pkgWatch` object telling the package where your local pixi.js build is (make sure it points to the .cjs bundle):

```json
{
    "pkgWatch": {
         "pixi.js": "relative/path/to/pixi.js/"
    }
}
```

* `npm run open -- --fiddle myfiddle`

An `index.js` file with a `const PIXI = window.PIXI || require('../relative/path/to/pixi.js')` will already be present. You can add code to experiment with PixiJS.

`npm run start` will run your fiddle in the local browser.
